integration-test/src/main/java/com/sequenceiq/it/cloudbreak/util/ssh/action/SshJClientActions.java

```diff
@@ -257,6 +257,23 @@ private void checkNoOutboundInternetTraffic(String instanceIp) {
         }
     }
 
+    public SdxTestDto checkKinitDuringFreeipaUpgrade(SdxTestDto testDto, List<InstanceGroupV4Response> instanceGroups, List<String> hostGroupNames) {
+        getSdxInstanceGroupIps(instanceGroups, hostGroupNames, false).stream().findFirst().ifPresent(ip -> checkKinitDuringFreeipaUpgrade(ip));
+        return testDto;
+    }
+
+    private void checkKinitDuringFreeipaUpgrade(String instanceIp) {
+        // TODO remove sleep 31 after unbound removal has been done, this is because unbound cached for30 sec the dns records
+        Pair<Integer, String> cmdOut = executeSshCommand(instanceIp,
+                "set -x sleep 31 && kdestroy && echo Password123! | KRB5_TRACE=/dev/stdout kinit -V fakemockuser0 && klist | grep fakemockuser0");
```

Comment:
you should sleep only if the kinit fails. it's not reasonable to have a 30 sec wait every time